src/wazuh_modules/main.c

```diff
@@ -16,13 +16,21 @@ static void wm_help();                  // Print help.
 static void wm_setup();                 // Setup function. Exits on error.
 static void wm_cleanup();               // Cleanup function, called on exiting.
 static void wm_handler(int signum);     // Action on signal.
+static void wm_signals_ignore();        // Ignore signals until the modules have started.
+static void wm_signals_configure();     // Configure signal handling.
 
 static int flag_foreground = 0;         // Running in foreground.
 
+static bool wm_sig_received = false;   // Flag to indicate if a signal is being handled.
+static pthread_mutex_t sig_lock = PTHREAD_MUTEX_INITIALIZER;
+
 // Main function
 
 int main(int argc, char **argv)
 {
+    // Ignore signals until the modules have started
+    wm_signals_ignore();
```

Comment:
Note: Performed some testing over this

If the module initialization takes too long or it hungs we are still able to kill the process, I don't know if this desirable or not but its working as intended

Reply:
I'm open to comments here. Ignoring the signals also avoids the `Termination` of the process when it receives a signal during the starting phase, but it might change the current behavior too much.

The real fix is in the mutex, I can revert this.